File: python_scripts/plot_antibiotic_comparison.R
#!/usr/bin/env Rscript
# Compare antibiotic usage between UCMC and ZCH
# Uses same styling as Figure 4

library(ggplot2)
library(dplyr)
library(tidyr)
library(gridExtra)
library(grid)
library(ggthemes)

# Set working directory
setwd("/home/david/projects/ZCH_UCMC_Manuscript")

# Create output directory
dir.create("revision_figures", showWarnings = FALSE)

cat("\n=== Loading Antibiotic Data ===\n")

# Load data
uc_data <- read.csv("UC_antibiotics_parsed.csv")
zch_data <- read.csv("ZCH_antibiotics_parsed.csv")

# Add location column
uc_data$Location <- "UCMC"
zch_data$Location <- "ZCH"

# Standardize MRN column names
colnames(uc_data)[colnames(uc_data) == "mrn_v2"] <- "MRN"
colnames(zch_data)[colnames(zch_data) == "mrn_v2_zju"] <- "MRN"

# Get antibiotic columns (exclude record_id, MRN, Location)
antibiotic_cols <- grep("^(Ampicillin|Penicillin|Nafcillin|Cefotaxime|Moxalactam|Ceftazidime|Ceftriaxone|Cefepime|Gentamicin|Tobramycin|Vancomycin|Fluconazole|Amphotericin|Meropenem|Azithromycin|Piperacillin)",
                        colnames(uc_data), value = TRUE)

# Separate week 1 and week 2 (which is actually week 3 data) columns
w1_cols <- grep("_w1$", antibiotic_cols, value = TRUE)
w2_cols <- grep("_w2$", antibiotic_cols, value = TRUE)

cat(sprintf("Week 1 columns: %d\n", length(w1_cols)))
cat(sprintf("Week 2 (week 3 data) columns: %d\n", length(w2_cols)))

# Create cumulative week 3 columns for both datasets
cat("\n=== Creating cumulative week 3 columns ===\n")
for (i in seq_along(w1_cols)) {
  w1_col <- w1_cols[i]
  # Find matching w2 column (replace _w1 with _w2)
  w2_col <- gsub("_w1$", "_w2", w1_col)

  if (w2_col %in% colnames(uc_data)) {
    # Create cumulative column name
    cumulative_col <- gsub("_w1$", "_cumulative_w3", w1_col)

    # Add cumulative columns to UC data
    uc_data[[cumulative_col]] <- uc_data[[w1_col]] + uc_data[[w2_col]]

    # Add cumulative columns to ZCH data
    zch_data[[cumulative_col]] <- zch_data[[w1_col]] + zch_data[[w2_col]]
  }
}

# Get updated antibiotic columns including cumulative
antibiotic_cols <- grep("^(Ampicillin|Penicillin|Nafcillin|Cefotaxime|Moxalactam|Ceftazidime|Ceftriaxone|Cefepime|Gentamicin|Tobramycin|Vancomycin|Fluconazole|Amphotericin|Meropenem|Azithromycin|Piperacillin)",
                        colnames(uc_data), value = TRUE)

# Now separate into week 1 and cumulative week 3 columns
w1_cols <- grep("_w1$", antibiotic_cols, value = TRUE)
cumulative_w3_cols <- grep("_cumulative_w3$", antibiotic_cols, value = TRUE)

cat(sprintf("Created %d cumulative week 3 columns\n", length(cumulative_w3_cols)))

# Combine datasets
combined_data <- bind_rows(
  uc_data[, c("MRN", "Location", antibiotic_cols)],
  zch_data[, c("MRN", "Location", antibiotic_cols)]
)

cat(sprintf("Combined data: %d UCMC + %d ZCH = %d total patients\n",
            nrow(uc_data), nrow(zch_data), nrow(combined_data)))

# Define color palette (same as Figure 4)
col <- c("#E69F00", "#56B4E9")  # UCMC (orange), ZCH (blue)

# Function to create individual antibiotic plot (Figure 4 style)
create_antibiotic_plot <- function(data, antibiotic, panel_label) {

  # Reshape data for this antibiotic
  # Keep ALL patients (including zeros) so both locations always appear
  plot_data <- data %>%
    select(MRN, Location, all_of(antibiotic)) %>%
    rename(days = all_of(antibiotic)) %>%
    filter(!is.na(days))

  if (sum(plot_data$days) == 0) {
    return(NULL)  # Skip if no usage at all
  }

  # For plotting, we'll still only show points for days > 0
  plot_data_points <- plot_data %>% filter(days > 0)

  # Calculate sample sizes (only count patients who received it)
  sample_counts <- plot_data_points %>%
    group_by(Location) %>%
    summarise(
      n = n(),
      max_days = max(days, na.rm = TRUE),
      .groups = 'drop'
    ) %>%
    mutate(y_pos = max_days * 1.3)  # Position above boxes

  # Calculate p-value (Wilcoxon test)
  ucmc_days <- plot_data %>% filter(Location == "UCMC") %>% pull(days)
  zch_days <- plot_data %>% filter(Location == "ZCH") %>% pull(days)

  # For the full comparison (including zeros from original data)
  ucmc_all <- data %>% filter(Location == "UCMC") %>% pull(!!antibiotic)
  zch_all <- data %>% filter(Location == "ZCH") %>% pull(!!antibiotic)

  if (length(ucmc_all) > 0 && length(zch_all) > 0) {
    test_result <- tryCatch({
      wilcox.test(ucmc_all, zch_all)
    }, error = function(e) {
      list(p.value = NA)
    })

    p_value <- test_result$p.value
    max_y <- max(plot_data$days, na.rm = TRUE)

    p_label <- ifelse(is.na(p_value), "",
                ifelse(p_value < 0.001, "p<0.001***",
                ifelse(p_value < 0.01, sprintf("p=%.3f**", p_value),
                ifelse(p_value < 0.05, sprintf("p=%.3f*", p_value),
                       sprintf("p=%.2f", p_value)))))

    pvalue_df <- data.frame(
      x = 1.5,
      y_pos = max_y * 1.5,
      p_label = p_label
    )
  } else {
    pvalue_df <- data.frame(x = numeric(0), y_pos = numeric(0), p_label = character(0))
  }

  # Get actual counts
  ucmc_count <- sample_counts %>% filter(Location == "UCMC") %>% pull(n)
  zch_count <- sample_counts %>% filter(Location == "ZCH") %>% pull(n)
  ucmc_count <- ifelse(length(ucmc_count) == 0, 0, ucmc_count)
  zch_count <- ifelse(length(zch_count) == 0, 0, zch_count)

  cat(sprintf("  %s: UCMC n=%d, ZCH n=%d, p=%s\n",
              antibiotic,
              ucmc_count,
              zch_count,
              if(nrow(pvalue_df) > 0) pvalue_df$p_label else "N/A"))

  # Clean antibiotic name for display
  # Convert _w1 to _wk1 and _cumulative_w3 to (cumulative wk3)
  clean_name <- gsub("_cumulative_w3", " (cumulative wk3)", antibiotic)
  clean_name <- gsub("_w1", " (wk1)", clean_name)
  clean_name <- gsub("_", " ", clean_name)

  # Special handling for Piperacillin Tazobactam - put on two lines
  if (grepl("Piperacillin", clean_name)) {
    clean_name <- gsub("Piperacillin Tazobactam", "Piperacillin\nTazobactam", clean_name)
  }

  # Create plot matching Figure 4 style
  # Use plot_data (includes zeros) for boxplot, but plot_data_points for individual points
  p <- ggplot(plot_data, aes(x = Location, y = days)) +
    geom_boxplot(lwd = 1, aes(color = factor(Location)),
                 fill = NA,
                 outlier.shape = NA) +  # Don't show outliers, we'll add points separately
    stat_summary(fun = mean, geom = "point", shape = 5, size = 4.8,
                data = plot_data_points) +  # Only show mean for non-zero
    scale_colour_manual(values = col) +
    geom_point(data = plot_data_points, size = 2.4,
              aes(color = factor(Location))) +  # Only show points for non-zero
    xlab(NULL) +
    ylab(clean_name) +
    scale_color_tableau() +
    theme_bw() +
    theme(
      axis.title.y = element_text(size = 20, face = "bold"),  # Antibiotic name - increased to 20
      axis.text.x = element_text(size = 18),  # Location labels (UCMC/ZCH)
      axis.text.y = element_text(size = 18),  # Y-axis tick labels
      legend.position = "none",
      panel.grid.major = element_line(color = "gray90"),
      panel.grid.minor = element_blank()
    ) +
    # Add sample count labels (increased size from 4 to 7)
    geom_text(data = sample_counts,
              aes(x = Location, y = y_pos, label = paste0("n=", n)),
              size = 7,
              fontface = "bold",
              inherit.aes = FALSE)

  # Add p-value if available (increased size from 4 to 7)
  if (nrow(pvalue_df) > 0 && pvalue_df$p_label != "") {
    p <- p + geom_text(data = pvalue_df,
                      aes(x = x, y = y_pos, label = p_label),
                      size = 7,
                      fontface = "bold",
                      inherit.aes = FALSE)
  }

  # Add panel label
  p <- p + ggtitle(paste0(panel_label, "."))

  return(p)
}

# Identify antibiotics with usage - separate by week
cat("\n=== Identifying antibiotics with usage ===\n")

# Week 1 antibiotics
w1_with_usage <- c()
for (antibiotic in w1_cols) {
  total_usage <- sum(combined_data[[antibiotic]] > 0, na.rm = TRUE)
  if (total_usage > 0) {
    w1_with_usage <- c(w1_with_usage, antibiotic)
  }
}

# Cumulative Week 3 antibiotics
w3_with_usage <- c()
for (antibiotic in cumulative_w3_cols) {
  total_usage <- sum(combined_data[[antibiotic]] > 0, na.rm = TRUE)
  if (total_usage > 0) {
    w3_with_usage <- c(w3_with_usage, antibiotic)
  }
}

cat(sprintf("Week 1: Found %d/%d antibiotics with usage\n",
            length(w1_with_usage), length(w1_cols)))
cat(sprintf("Cumulative Week 3: Found %d/%d antibiotics with usage\n",
            length(w3_with_usage), length(cumulative_w3_cols)))

# Generate plots for Week 1
cat("\n=== Generating Week 1 plots ===\n")
w1_plot_list <- list()
w1_panel_labels <- letters[1:length(w1_with_usage)]

for (i in seq_along(w1_with_usage)) {
  antibiotic <- w1_with_usage[i]
  panel_label <- w1_panel_labels[i]

  p <- create_antibiotic_plot(combined_data, antibiotic, panel_label)
  if (!is.null(p)) {
    w1_plot_list[[length(w1_plot_list) + 1]] <- p
  }
}

# Generate plots for Cumulative Week 3
cat("\n=== Generating Cumulative Week 3 plots ===\n")
w3_plot_list <- list()
w3_panel_labels <- letters[1:length(w3_with_usage)]

for (i in seq_along(w3_with_usage)) {
  antibiotic <- w3_with_usage[i]
  panel_label <- w3_panel_labels[i]

  p <- create_antibiotic_plot(combined_data, antibiotic, panel_label)
  if (!is.null(p)) {
    w3_plot_list[[length(w3_plot_list) + 1]] <- p
  }
}

# Create Week 1 figure
cat("\n=== Creating Week 1 figure ===\n")

n_plots_w1 <- length(w1_plot_list)
ncols <- 4
nrows_w1 <- ceiling(n_plots_w1 / ncols)

w1_fig <- arrangeGrob(
  grobs = w1_plot_list,
  ncol = ncols,
  nrow = nrows_w1,
  top = textGrob("Supplementary Figure A. Antibiotic Usage Comparison (Week 1): UCMC vs ZCH\n",
                 gp = gpar(fontsize = 24, fontface = "bold")),
  padding = unit(0.8, "line"),
  widths = unit(rep(3.78, ncols), "in"),
  heights = unit(rep(3.5, nrows_w1), "in")
)

# Calculate figure dimensions for Week 1
fig_width_w1 <- ncols * 4.5
fig_height_w1 <- nrows_w1 * 4

# Save Week 1 figure
ggsave("revision_figures/Antibiotic_Comparison_UCMC_vs_ZCH_Week1.pdf",
       plot = w1_fig,
       width = fig_width_w1,
       height = fig_height_w1,
       limitsize = FALSE)

cat("✓ Saved Week 1 antibiotic comparison figure\n")

# Create Cumulative Week 3 figure
cat("\n=== Creating Cumulative Week 3 figure ===\n")

n_plots_w3 <- length(w3_plot_list)
nrows_w3 <- ceiling(n_plots_w3 / ncols)

w3_fig <- arrangeGrob(
  grobs = w3_plot_list,
  ncol = ncols,
  nrow = nrows_w3,
  top = textGrob("Supplementary Figure B. Antibiotic Usage Comparison (Cumulative Week 3): UCMC vs ZCH\n",
                 gp = gpar(fontsize = 24, fontface = "bold")),
  padding = unit(0.8, "line"),
  widths = unit(rep(3.78, ncols), "in"),
  heights = unit(rep(3.5, nrows_w3), "in")
)

# Calculate figure dimensions for Week 3
fig_width_w3 <- ncols * 4.5
fig_height_w3 <- nrows_w3 * 4

# Save Week 3 figure
ggsave("revision_figures/Antibiotic_Comparison_UCMC_vs_ZCH_CumulativeWeek3.pdf",
       plot = w3_fig,
       width = fig_width_w3,
       height = fig_height_w3,
       limitsize = FALSE)

cat("✓ Saved Cumulative Week 3 antibiotic comparison figure\n")

# ============================================================================
# Generate statistics table
# ============================================================================

cat("\n=== Calculating statistics ===\n")

stats_results <- data.frame()

# Combine week 1 and cumulative week 3 antibiotics for statistics
all_antibiotics_for_stats <- c(w1_with_usage, w3_with_usage)

for (antibiotic in all_antibiotics_for_stats) {
  ucmc_data <- combined_data %>% filter(Location == "UCMC") %>% pull(!!antibiotic)
  zch_data <- combined_data %>% filter(Location == "ZCH") %>% pull(!!antibiotic)

  # Prevalence
  ucmc_prev <- sum(ucmc_data > 0) / length(ucmc_data) * 100
  zch_prev <- sum(zch_data > 0) / length(zch_data) * 100

  # Mean days (among those who received)
  ucmc_mean <- mean(ucmc_data[ucmc_data > 0])
  zch_mean <- mean(zch_data[zch_data > 0])

  # Count
  ucmc_n <- sum(ucmc_data > 0)
  zch_n <- sum(zch_data > 0)

  # Wilcoxon test
  if (length(ucmc_data) > 0 && length(zch_data) > 0) {
    test_result <- tryCatch({
      wilcox.test(ucmc_data, zch_data)
    }, error = function(e) {
      list(p.value = NA)
    })
    p_value <- test_result$p.value
  } else {
    p_value <- NA
  }

  stats_results <- rbind(stats_results, data.frame(
    Antibiotic = antibiotic,
    UCMC_N = ucmc_n,
    UCMC_Prevalence_Percent = ucmc_prev,
    UCMC_Mean_Days = ifelse(is.nan(ucmc_mean), 0, ucmc_mean),
    ZCH_N = zch_n,
    ZCH_Prevalence_Percent = zch_prev,
    ZCH_Mean_Days = ifelse(is.nan(zch_mean), 0, zch_mean),
    P_value = p_value,
    Significant = ifelse(!is.na(p_value) && p_value < 0.05, "Yes", "No")
  ))
}

# Sort by p-value
stats_results <- stats_results %>% arrange(P_value)

# Save statistics
write.csv(stats_results,
          "revision_figures/Antibiotic_Statistics_UCMC_vs_ZCH.csv",
          row.names = FALSE)

cat("✓ Saved statistics table\n")

# Print summary
cat("\n")
cat(paste(rep("=", 70), collapse = ""))
cat("\n")
cat("ANTIBIOTIC COMPARISON COMPLETE\n")
cat(paste(rep("=", 70), collapse = ""))
cat("\n")
cat(sprintf("\nTotal patients: %d (%d UCMC, %d ZCH)\n",
            nrow(combined_data),
            sum(combined_data$Location == "UCMC"),
            sum(combined_data$Location == "ZCH")))
cat(sprintf("Week 1 antibiotics with usage: %d/%d\n",
            length(w1_with_usage), length(w1_cols)))
cat(sprintf("Cumulative Week 3 antibiotics with usage: %d/%d\n",
            length(w3_with_usage), length(cumulative_w3_cols)))
cat(sprintf("Significant differences (p<0.05): %d\n",
            sum(stats_results$P_value < 0.05, na.rm = TRUE)))

cat("\nGenerated files:\n")
cat("  - revision_figures/Antibiotic_Comparison_UCMC_vs_ZCH_Week1.pdf\n")
cat("  - revision_figures/Antibiotic_Comparison_UCMC_vs_ZCH_CumulativeWeek3.pdf\n")
cat("  - revision_figures/Antibiotic_Statistics_UCMC_vs_ZCH.csv\n")

cat("\nTop significant differences:\n")
sig_results <- stats_results %>%
  filter(P_value < 0.05) %>%
  head(10)
for (i in 1:min(10, nrow(sig_results))) {
  cat(sprintf("  %s: UCMC %.1f%% vs ZCH %.1f%% (p=%.4f)\n",
              sig_results$Antibiotic[i],
              sig_results$UCMC_Prevalence_Percent[i],
              sig_results$ZCH_Prevalence_Percent[i],
              sig_results$P_value[i]))
}

cat(paste(rep("=", 70), collapse = ""))
cat("\n")

File: scripts/python/visualization/plot_antibiotic_comparison.py
#!/usr/bin/env python3
"""
Compare antibiotic usage between UCMC and ZCH.
Creates boxplots showing days of each antibiotic per patient.
"""

import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
from scipy import stats
from pathlib import Path

# Set style
sns.set_style("whitegrid")
plt.rcParams['figure.dpi'] = 300

def load_and_combine_data():
    """Load both antibiotic files and combine with location indicator."""

    print("Loading antibiotic data...")

    # Load UC data
    uc_df = pd.read_csv('UC_antibiotics_parsed.csv')
    uc_df['Location'] = 'UCMC'

    # Load ZCH data
    zch_df = pd.read_csv('ZCH_antibiotics_parsed.csv')
    zch_df['Location'] = 'ZCH'

    # Get antibiotic columns (exclude record_id, mrn, and Location)
    uc_abx_cols = [col for col in uc_df.columns if any(abx in col for abx in
                   ['Ampicillin', 'Penicillin', 'Nafcillin', 'Cefotaxime', 'Moxalactam',
                    'Ceftazidime', 'Ceftriaxone', 'Cefepime', 'Gentamicin', 'Tobramycin',
                    'Vancomycin', 'Fluconazole', 'Amphotericin', 'Meropenem', 'Azithromycin',
                    'Piperacillin'])]

    # Standardize column names between datasets
    # UC has mrn_v2, ZCH has mrn_v2_zju
    uc_df = uc_df.rename(columns={'mrn_v2': 'MRN'})
    zch_df = zch_df.rename(columns={'mrn_v2_zju': 'MRN'})

    # Select relevant columns for combining
    uc_cols = ['MRN', 'Location'] + uc_abx_cols
    zch_cols = ['MRN', 'Location'] + uc_abx_cols  # Use same column list

    # Combine datasets
    combined_df = pd.concat([uc_df[uc_cols], zch_df[zch_cols]], ignore_index=True)

    print(f"Combined data: {len(uc_df)} UCMC patients + {len(zch_df)} ZCH patients = {len(combined_df)} total")

    return combined_df, uc_abx_cols

def calculate_statistics(data, antibiotic_cols):
    """Calculate statistics comparing UCMC vs ZCH for each antibiotic."""

    stats_results = []

    for col in antibiotic_cols:
        ucmc_data = data[data['Location'] == 'UCMC'][col]
        zch_data = data[data['Location'] == 'ZCH'][col]

        # Remove zeros for statistics (only count patients who received the antibiotic)
        ucmc_nonzero = ucmc_data[ucmc_data > 0]
        zch_nonzero = zch_data[zch_data > 0]

        # Calculate prevalence (% of patients who received this antibiotic)
        ucmc_prevalence = (ucmc_data > 0).sum() / len(ucmc_data) * 100
        zch_prevalence = (zch_data > 0).sum() / len(zch_data) * 100

        # Wilcoxon rank-sum test (Mann-Whitney U) - comparing all values including zeros
        if len(ucmc_data) > 0 and len(zch_data) > 0:
            stat, pval = stats.mannwhitneyu(ucmc_data, zch_data, alternative='two-sided')
        else:
            pval = np.nan

        # Mean days among those who received the antibiotic
        ucmc_mean = ucmc_nonzero.mean() if len(ucmc_nonzero) > 0 else 0
        zch_mean = zch_nonzero.mean() if len(zch_nonzero) > 0 else 0

        stats_results.append({
            'Antibiotic': col,
            'UCMC_N': len(ucmc_nonzero),
            'UCMC_Prevalence_%': ucmc_prevalence,
            'UCMC_Mean_Days': ucmc_mean,
            'ZCH_N': len(zch_nonzero),
            'ZCH_Prevalence_%': zch_prevalence,
            'ZCH_Mean_Days': zch_mean,
            'P_value': pval,
            'Significant': 'Yes' if pval < 0.05 else 'No'
        })

    stats_df = pd.DataFrame(stats_results)

    # Sort by p-value
    stats_df = stats_df.sort_values('P_value')

    return stats_df

def create_boxplot_grid(data, antibiotic_cols, stats_df, output_dir='revision_figures'):
    """Create comprehensive boxplot grid for all antibiotics."""

    Path(output_dir).mkdir(exist_ok=True)

    # Filter to antibiotics that have at least some usage
    antibiotics_with_usage = []
    for col in antibiotic_cols:
        if data[col].sum() > 0:  # At least some patients received this
            antibiotics_with_usage.append(col)

    print(f"\nAntibiotics with usage: {len(antibiotics_with_usage)}/{len(antibiotic_cols)}")

    if len(antibiotics_with_usage) == 0:
        print("No antibiotics with usage found!")
        return

    # Calculate grid dimensions
    n_antibiotics = len(antibiotics_with_usage)
    ncols = 4
    nrows = int(np.ceil(n_antibiotics / ncols))

    # Create figure
    fig, axes = plt.subplots(nrows, ncols, figsize=(16, 4*nrows))
    axes = axes.flatten() if n_antibiotics > 1 else [axes]

    # Color palette
    colors = {'UCMC': '#E69F00', 'ZCH': '#56B4E9'}

    for idx, antibiotic in enumerate(antibiotics_with_usage):
        ax = axes[idx]

        # Prepare data for plotting
        plot_data = data[['Location', antibiotic]].copy()
        plot_data = plot_data[plot_data[antibiotic] > 0]  # Only show patients who received it

        if len(plot_data) == 0:
            ax.text(0.5, 0.5, 'No usage', ha='center', va='center')
            ax.set_title(antibiotic.replace('_', ' '))
            ax.axis('off')
            continue

        # Create boxplot
        positions = [0, 1]
        box_data = [plot_data[plot_data['Location'] == 'UCMC'][antibiotic],
                    plot_data[plot_data['Location'] == 'ZCH'][antibiotic]]

        bp = ax.boxplot(box_data, positions=positions, widths=0.6,
                        patch_artist=True, showfliers=False)

        # Color boxes
        for patch, location in zip(bp['boxes'], ['UCMC', 'ZCH']):
            patch.set_facecolor(colors[location])
            patch.set_alpha(0.7)

        # Add individual points with jitter
        for pos, location in enumerate(['UCMC', 'ZCH']):
            location_data = plot_data[plot_data['Location'] == location][antibiotic]
            if len(location_data) > 0:
                # Add jitter to x-axis
                x_jitter = np.random.normal(pos, 0.04, size=len(location_data))
                ax.scatter(x_jitter, location_data, alpha=0.6, s=30,
                          color=colors[location], edgecolors='black', linewidth=0.5)

        # Get p-value for this antibiotic
        pval = stats_df[stats_df['Antibiotic'] == antibiotic]['P_value'].values[0]

        # Format p-value
        if pd.notna(pval):
            if pval < 0.001:
                pval_str = 'p<0.001***'
            elif pval < 0.01:
                pval_str = f'p={pval:.3f}**'
            elif pval < 0.05:
                pval_str = f'p={pval:.3f}*'
            else:
                pval_str = f'p={pval:.3f}'
        else:
            pval_str = 'p=N/A'

        # Add p-value as text
        y_max = plot_data[antibiotic].max()
        ax.text(0.5, y_max * 1.15, pval_str, ha='center', va='bottom',
               fontsize=10, fontweight='bold')

        # Styling
        ax.set_xticks([0, 1])
        ax.set_xticklabels(['UCMC', 'ZCH'], fontsize=11)
        ax.set_ylabel('Days', fontsize=11)
        ax.set_title(antibiotic.replace('_', ' '), fontsize=12, fontweight='bold')
        ax.spines['top'].set_visible(False)
        ax.spines['right'].set_visible(False)
        ax.grid(axis='y', alpha=0.3)

    # Hide unused subplots
    for idx in range(len(antibiotics_with_usage), len(axes)):
        axes[idx].axis('off')

    plt.tight_layout()

    # Save figure
    output_file = f'{output_dir}/Antibiotic_Comparison_UCMC_vs_ZCH.pdf'
    plt.savefig(output_file, bbox_inches='tight', dpi=300)
    print(f"\n✓ Saved boxplot figure: {output_file}")

    plt.close()

def create_summary_figure(stats_df, output_dir='revision_figures'):
    """Create summary figure showing significant differences."""

    Path(output_dir).mkdir(exist_ok=True)

    # Filter to antibiotics with p < 0.05
    sig_antibiotics = stats_df[stats_df['P_value'] < 0.05].copy()

    if len(sig_antibiotics) == 0:
        print("\nNo significant differences found between UCMC and ZCH")
        return

    print(f"\n{len(sig_antibiotics)} antibiotics with significant differences (p<0.05)")

    # Create bar plot
    fig, ax = plt.subplots(figsize=(10, max(6, len(sig_antibiotics)*0.4)))

    x = np.arange(len(sig_antibiotics))
    width = 0.35

    ucmc_prev = sig_antibiotics['UCMC_Prevalence_%'].values
    zch_prev = sig_antibiotics['ZCH_Prevalence_%'].values

    ax.barh(x - width/2, ucmc_prev, width, label='UCMC', color='#E69F00', alpha=0.8)
    ax.barh(x + width/2, zch_prev, width, label='ZCH', color='#56B4E9', alpha=0.8)

    ax.set_yticks(x)
    ax.set_yticklabels([abx.replace('_', ' ') for abx in sig_antibiotics['Antibiotic']],
                       fontsize=10)
    ax.set_xlabel('Prevalence (%)', fontsize=12)
    ax.set_title('Antibiotics with Significant Differences in Usage\n(p < 0.05)',
                fontsize=14, fontweight='bold')
    ax.legend()
    ax.spines['top'].set_visible(False)
    ax.spines['right'].set_visible(False)
    ax.grid(axis='x', alpha=0.3)

    plt.tight_layout()

    output_file = f'{output_dir}/Significant_Antibiotic_Differences.pdf'
    plt.savefig(output_file, bbox_inches='tight', dpi=300)
    print(f"✓ Saved significance summary: {output_file}")

    plt.close()

def main():
    """Main function."""

    print("="*70)
    print("ANTIBIOTIC USAGE COMPARISON: UCMC vs ZCH")
    print("="*70)

    # Load and combine data
    combined_df, antibiotic_cols = load_and_combine_data()

    # Calculate statistics
    print("\nCalculating statistics...")
    stats_df = calculate_statistics(combined_df, antibiotic_cols)

    # Save statistics
    stats_file = 'revision_figures/Antibiotic_Statistics_UCMC_vs_ZCH.csv'
    stats_df.to_csv(stats_file, index=False)
    print(f"✓ Saved statistics: {stats_file}")

    # Print summary
    print("\n" + "="*70)
    print("SUMMARY STATISTICS")
    print("="*70)
    print(f"\nAntibiotics with significant differences (p<0.05): {(stats_df['P_value'] < 0.05).sum()}")

    if (stats_df['P_value'] < 0.05).any():
        print("\nSignificant antibiotics:")
        sig_df = stats_df[stats_df['P_value'] < 0.05][['Antibiotic', 'UCMC_Prevalence_%',
                                                         'ZCH_Prevalence_%', 'P_value']]
        print(sig_df.to_string(index=False))

    # Create visualizations
    print("\n" + "="*70)
    print("Creating visualizations...")
    print("="*70)

    create_boxplot_grid(combined_df, antibiotic_cols, stats_df)
    create_summary_figure(stats_df)

    print("\n" + "="*70)
    print("COMPLETE")
    print("="*70)
    print("\nGenerated files:")
    print("  - revision_figures/Antibiotic_Comparison_UCMC_vs_ZCH.pdf")
    print("  - revision_figures/Antibiotic_Statistics_UCMC_vs_ZCH.csv")
    print("  - revision_figures/Significant_Antibiotic_Differences.pdf")

if __name__ == '__main__':
    main()
